lib/iris/tests/__init__.py

```diff
@@ -695,93 +702,99 @@ def _assert_graphic(self):
             result_fname = os.path.join(image_output_directory,
                                         'result-' + unique_id + '.png')
 
-            if not os.path.isdir(os.path.dirname(result_fname)):
+            if not os.path.isdir(image_output_directory):
                 # Handle race-condition where the directories are
                 # created sometime between the check above and the
                 # creation attempt below.
                 try:
-                    os.makedirs(os.path.dirname(result_fname))
+                    os.makedirs(image_output_directory)
                 except OSError as err:
                     # Don't care about "File exists"
                     if err.errno != 17:
                         raise
 
-            def _save_figure_hash():
-                plt.gcf().savefig(result_fname)
-                # Determine the test result image hash using sha1.
-                with open(result_fname, 'rb') as fi:
-                    sha1 = hashlib.sha1(fi.read())
-                return sha1
-
```

Comment:
This hot-fix didn't work and needs to be purged.

At this point the plot was **already** wrong i.e. another thread had already clobbered either the plot title or axes title/s, so re-saving the clobbered plot image was never going to yield the correct sha.